docs/design/index.md

```diff
@@ -38,6 +38,10 @@ The downside of the microautonomy architecture is that the computational perform
 
     Under Construction
 
+### Autoware architecture
```

Comment:
Mm, this should be at least after the `Autoware concepts` section.
And it might be placed under the `autoware-interfaces/` directory.

I'd like to ask other people about where to put these documents.

@xmfcx @mitsudome-r @isamu-takagi @LalithVipulananthan Where do you think is the best place?

Reply:
@kenji-miyake @TakaHoribe 
> And it might be placed under the autoware-interfaces/ directory.

I think `design/autoware-architecture/<component-name>/` is an overview and `design/autoware-interfaces/components/<component-name>/` is a detail. So the architecture focuses on what components and modules are, how they are connected, what the purpose is, and so on. On the other hand, Interface describes input and output for each components or module.

For example, in the architecture section, there are perception and planning components. And they are sending and receiving data of predicted objects to avoid or stop. Here, a predicted object is just a predicted object, it`s data structure is not explained unless it is necessary. The interface section describes that predicted object has header, uuid, classification, kinematics, etc. like [this page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/components/planning/#inputs).

Also, the component interface is for the public interface. The internal messages are described elsewhere, such as in the README.

Reply:
@isamu-takagi Thank you for your comment.
So is this good?

Autoware architecture
- target readers: beginner developers/users
- purpose: give them a high-level understanding

Autoware interfaces
- target readers: accustomed developers
- purpose: give them the information of how-to-develop

Reply:
I think this is a good way to organise the content.  👍🏾

Reply:
I updated the document. How's this?

@LalithVipulananthan I would appreciate it if you could make modifications as needed.

Reply:
@TakaHoribe Looks fine to me!